File: blog/src/main/kotlin/com/liux/blog/bean/po/Url.kt
package com.liux.blog.bean.po

import com.liux.blog.bean.NoArgsConstructor
import java.util.*

// 短链接数据
@NoArgsConstructor
class Url(
    // ID
    var id: Int? = null,
    // 键
    var key: String? = null,
    // 链接地址
    var url: String? = null,
    // 描述信息
    var description: String? = null,
    // 作者ID
    var authorId: Int? = null,
    // 访问次数
    var views: Int? = null,
    // 访问次数
    var totalViews: Int? = null,
    // 创建时间
    var createTime: Date? = null,
    // 更新时间
    var updateTime: Date? = null,
    // 状态
    var status: Int? = null,

    // 作者
    var author: User? = null
)

File: blog/bean/api_vo/url.go
package api_vo

import (
	"blog/bean/po"
	"time"
)

type UrlVO struct {
	Id          int        `json:"id"`
	Key         string     `json:"key"`
	Url         string     `json:"url"`
	Description string     `json:"description"`
	AuthorId    int        `json:"authorId"`
	Views       int        `json:"views"`
	TotalViews  int        `json:"totalViews"`
	AuthorName  string     `json:"authorName"`
	Status      int        `json:"status"`
	CreateTime  time.Time  `json:"createTime"`
	UpdateTime  *time.Time `json:"updateTime"`
}

func (v *UrlVO) From(url po.Url) {
	v.Id = url.Id
	v.Key = url.Key
	v.Url = url.Url
	v.Description = url.Description
	v.AuthorId = url.AuthorId
	v.Views = url.Views
	v.TotalViews = url.TotalViews
	v.AuthorName = url.Author.Nickname
	v.Status = url.Status
	v.CreateTime = url.CreateTime
	v.UpdateTime = url.UpdateTime
}
